internal/archive/archive_test.go

```diff
@@ -492,44 +519,122 @@ func read(r io.Reader) string {
 
 func (s *S) TestRealArchive(c *C) {
 	if !*realArchiveFlag {
 		c.Skip("--real-archive not provided")
 	}
-	for _, release := range ubuntuReleases {
-		for _, arch := range elfToDebArch {
-			s.testOpenArchiveArch(c, release, arch)
+	allArch := make([]string, 0, len(elfToDebArch))
+	for _, arch := range elfToDebArch {
+		allArch = append(allArch, arch)
+	}
+	for _, test := range realArchiveTests {
+		if test.pro != "" && !*proArchiveFlag {
+			// --pro-archive is not provided. Ignore test.
+			continue
+		}
+		if len(test.architectures) == 0 {
+			test.architectures = allArch
+		}
+		for _, arch := range test.architectures {
+			s.testOpenArchiveArch(c, test, arch)
 		}
 	}
 }
 
-type ubuntuRelease struct {
+type realArchiveTest struct {
 	name           string
 	version        string
+	suites         []string
+	components     []string
+	pro            string
 	archivePubKeys []*packet.PublicKey
+	architectures  []string
+	pkg            string
+	binPath        string
+	copyrightText  string
 }
 
-var ubuntuReleases = []ubuntuRelease{{
-	name:    "focal",
-	version: "20.04",
-	archivePubKeys: []*packet.PublicKey{
-		keyUbuntu2018.PubKey,
-	},
+var realArchiveTests = []realArchiveTest{{
+	name:           "focal",
+	version:        "20.04",
+	suites:         []string{"focal"},
+	components:     []string{"main", "universe"},
+	archivePubKeys: []*packet.PublicKey{keyUbuntu2018.PubKey},
+	pkg:            "hostname",
+	binPath:        "/bin/hostname",
+	copyrightText:  "This package was written by Peter Tobias <tobias@et-inf.fho-emden.de>",
 }, {
-	name:    "jammy",
-	version: "22.04",
-	archivePubKeys: []*packet.PublicKey{
-		keyUbuntu2018.PubKey,
-	},
+	name:           "jammy",
+	version:        "22.04",
+	suites:         []string{"jammy"},
+	components:     []string{"main", "universe"},
+	archivePubKeys: []*packet.PublicKey{keyUbuntu2018.PubKey},
+	pkg:            "hostname",
+	binPath:        "/bin/hostname",
+	copyrightText:  "This package was written by Peter Tobias <tobias@et-inf.fho-emden.de>",
 }, {
-	name:    "noble",
-	version: "24.04",
-	archivePubKeys: []*packet.PublicKey{
-		keyUbuntu2018.PubKey,
-	},
+	name:           "noble",
+	version:        "24.04",
+	suites:         []string{"noble"},
+	components:     []string{"main", "universe"},
+	archivePubKeys: []*packet.PublicKey{keyUbuntu2018.PubKey},
+	pkg:            "hostname",
+	binPath:        "/usr/bin/hostname",
+	copyrightText:  "This package was written by Peter Tobias <tobias@et-inf.fho-emden.de>",
+}, {
+	name:           "fips",
+	version:        "20.04",
+	suites:         []string{"focal"},
+	components:     []string{"main"},
+	pro:            "fips",
+	archivePubKeys: []*packet.PublicKey{keyUbuntuFIPSv1.PubKey},
+	architectures:  []string{"amd64"},
+	pkg:            "openssh-client",
+	binPath:        "/usr/bin/ssh",
+	copyrightText:  "1995 Tatu Ylonen <ylo@cs.hut.fi>, Espoo, Finland",
+}, {
```

Comment:
Let's have a look if we can test these for `jammy` as well.

Reply:
Jammy does not support `fips` yet. It does support `fips-updates` though. Focal seems to be a better choice.